pylmod/membership.py

```diff
@@ -18,7 +19,90 @@ def __init__(
             self,
             cert,
             urlbase='https://learning-modules.mit.edu:8443/',
+            cuuid=None
     ):
         super(Membership, self).__init__(cert, urlbase)
         # Add service base
         self.urlbase += 'service/membership/'
+        self.course_id = None
+        if cuuid is not None:
+            self.course_id = self.get_course_id(cuuid)
+
+    def get_course_id(self, course_uuid):
+        """Get course id based on uuid.
+
+        Args:
+            uuid (str): course uuid, i.e. /project/mitxdemosite
+
+        Raises:
+            PyLmodUnexpectedData: No course data was returned.
+            requests.RequestException: Exception connection error
+
+        Returns:
+            int: numeric course id
+
+        """
+        course_data = self.get(
+            'courseguide/course?uuid={uuid}'.format(
+                uuid=course_uuid or self.course_id
+            ),
+            params=None
+        )
+        try:
+            return course_data['response']['docs'][0]['id']
+        except Exception:
```

Comment:
Weird that this is passing tests, I guess flakes doesn't check for broad exceptions.  This should only except on specific errors, i.e. `TypeError` when the data isn't a dictionary and `KeyError` when it is a dictionary but the key doesn't exist.  You will probably want a test that exercises both.